src/NServiceBus.Core/Recoverability/SecondLevelRetries/SecondLevelRetriesBehavior.cs

```diff
@@ -15,16 +15,40 @@ namespace NServiceBus
 
     class SecondLevelRetriesBehavior : Behavior<TransportReceiveContext>
     {
-        public SecondLevelRetriesBehavior(IPipelineBase<RoutingContext> dispatchPipeline, SecondLevelRetryPolicy retryPolicy, BusNotifications notifications, string localAddress)
+        public SecondLevelRetriesBehavior(IPipelineBase<RoutingContext> dispatchPipeline, SecondLevelRetryPolicy retryPolicy, BusNotifications notifications, string localAddress, SlrStatusStorage slrStatusStorage)
         {
             this.dispatchPipeline = dispatchPipeline;
             this.retryPolicy = retryPolicy;
             this.notifications = notifications;
             this.localAddress = localAddress;
+            this.slrStatusStorage = slrStatusStorage;
         }
 
         public override async Task Invoke(TransportReceiveContext context, Func<Task> next)
         {
+            Exception exception;
+            var uniqueMessageId = PipelineInfo.Name + context.Message.MessageId;
+
+            if (slrStatusStorage.TryGetException(uniqueMessageId, out exception))
+            {
+                slrStatusStorage.ClearException(uniqueMessageId);
```

Comment:
@tmasternak 

> I think that's the expected behavior.

I think it would make sense to prove what you think with an acceptance test or a component test which sets up part of the pipeline to proof that. Thoughts?